addons/website_quote/controllers/main.py

```diff
@@ -189,8 +189,11 @@ def payment_transaction_token(self, acquirer_id, order_id, token):
                 'partner_id': Order.partner_id.id,
                 'reference': PaymentTransaction.get_next_reference(Order.name),
                 'sale_order_id': Order.id,
-                'callback_eval': "self.sale_order_id._confirm_online_quote(self)"
+                'callback_ir_model_id': request.env['ir.model'].sudo().search([('model', '=', Order._name)], limit=1).id,
+                'callback_res_id': Order.id,
+                'callback_method': '_confirm_online_quote',
             })
+            Transaction.write({'callback_hash': Transaction._generate_callback_hash()})
```

Comment:
Shouldn't this be done in a create override on `payment.transaction`? This is a matter of opinion since technically correct in both cases, but it seems to me that having to create my transaction in 2 operations is not obvious & a potential PITA.

Maybe we could also add a check to see if the model_id/res_id tuple actually exists in the database. Just a thought. Bonus if we check that the callback function exists as well. Should be quite easy to do with an `@api.constrains`.

@tde-banana-odoo 
I'd like your opinion here.